src/component/app/RegisteredUserRoute.tsx

```diff
@@ -19,15 +19,16 @@ export function RegisteredUserRoute(props: RegisteredUserRouteProps) {
 
   if (userLoading) return <span>Loading user...</span>;
 
-  if (userError) return <Navigate to={"../signin"} />;
-
-  if (!user) {
-    return <Navigate to={"../profile"} />;
+  const returnTo = window.location.pathname + window.location.search;
+  if (userError) {
+    return <Navigate to="/signin" state={{ returnTo }} />;
+  } else if (!user) {
+    return <Navigate to={"/profile"} state={{ returnTo }} />;
   }
 
   return (
     <Routes>
-      <Route path="joinventure" element={<JoinVenture />} />
+      <Route path="invite" element={<JoinVenture />} />
```

Comment:
No longer use both `/joinventure` and `/invite`. Instead we only link to `/invite` in the invite email.